### PR TITLE
Baseline cnn

### PR DESCRIPTION
**Pull request by BreziTasbi**
This branch is meant to provide a baseline model structure in order to have something to compete against.

I shaped the code so it can be easily re-used for other networks.

Basically data_manager.py contains everything to load the data from the looking for niftis to the dataset class and its preprocessing. (the expected nifti dataset can be created with "convert_dataset_dcm2nii.py")

It is used by both "train.py" which performs training and test accuracy on a validation dataset (use it with --num_epoch 0 if you want testing only) and by "submission.py" which perform inference on the test dataset and return the appropriate csv.

The model in itself is hidden in "classifier_Networks.py", I think adding any other network we want to use here is a good idea to keep the global code structure clean.

## Details

Input:
- one NIfTI volume

Output: prediction
- 3 diseases (incl. 2 diseases with separate labels depending on laterality, which makes a total of 5 conditions) = 5 conditions
- 3 degrees of severity
- 5 vertebral levels
TOTAL: 75

### CODE REVIEW

File: classifier_Networks.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import torch
```
Comment (valosekj) [inline classifier_Networks.py:6]: Please add a brief script description. This applies also to your other scripts.

File: convert_dataset_dcm2nii.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+for root, dirs, files in os.walk(input_folder):
+    for file in files:
+        if file.endswith(".dcm"):
+            if root not in processed_folders:
+                processed_folders.append(root)
+
+                root_nifti = root.replace(f"{dataset_type}_images", f"{dataset_type}_nifti")
+                if not os.path.exists(root_nifti):
+                    os.makedirs(root_nifti)
+                subprocess.run(f'dcm2niix -o {root_nifti} -z y {root}', shell=True)
```
Comment (valosekj) [inline convert_dataset_dcm2nii.py:36]: I think that also other team members (@Nilser3, @KaterinaKrejci231054) are working on dicom to nii conversion --> try to synchronize your efforts to reduce duplications

File: data_manager.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    SqueezeDim,
+    RandRotate,
+    RandSimulateLowResolution,
+    Resize,
+    CenterSpatialCrop,
+)
+import torch
+import subprocess
+
+device = torch.device("cuda:3" if torch.cuda.is_available() else "cpu")
```
Comment (valosekj) [inline data_manager.py]: not sure hardcoding `cuda:3` is a good idea. maybe it could be an input arg?

Comment (BreziTasbi) [reply data_manager.py]: indeed, i should change this asap

**Comment by jcohenadad** [comment]
@simonqueric what should we do with this PR?

**Comment by simonqueric** [comment]
> @simonqueric what should we do with this PR?

We should close this PR.
I decided to open another PR for baseline experiments (#19 ) because this one (#2) wasn't taking account a lot of relevent information about contrasts, resolutions and evaluation metric.
